tpmutil/encoding.go

```diff
@@ -54,6 +54,12 @@ func packedSize(elts ...interface{}) (int, error) {
 		v := reflect.ValueOf(e)
 		switch v.Kind() {
 		case reflect.Ptr:
+			// Check if it's a TPMSelfMarshaler
+			marshaler, ok := e.(TPMSelfMarshaler)
```

Comment:
Check this outside of the switch, regardless of e's reflect kind.
Same for all other places where it's used